pkg/kclient/deployments.go

```diff
@@ -0,0 +1,37 @@
+package kclient
+
+import (
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CreateDeployment creates a deployment based on the given pod
+func (c *Client) CreateDeployment(pod *corev1.Pod) (*appsv1.Deployment, error) {
+
+	replicas := int32(1)
+	deployment := appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
```

Comment:
We could extract `TypeMeta` and `ObjectMeta` in a function as its being using in tests as well e.g. https://github.com/openshift/odo/pull/2509/files#diff-2ee5055e18886da896e73c5cc2b7aa8fR72

Reply:
While we can create functions I'm wondering whether it would reduce readability and whether there are really that many cases where the function would be more helpful than just creating the struct. If you have some good examples, can you please share?

Reply:
Possibly the readability could reduce hence maybe its fine to keep it as is.